core/src/test/scala/unit/kafka/admin/PreferredReplicaLeaderElectionCommandTest.scala

```diff
@@ -0,0 +1,341 @@
+    try {
+      PreferredReplicaLeaderElectionCommand.run(Array(
+        "--bootstrap-server", bootstrapServer(),
+        "--path-to-json-file", jsonFile.getAbsolutePath))
+    } catch {
+      case e: AdminCommandFailedException =>
+        val suppressed = e.getSuppressed()(0)
+        assertTrue(suppressed.isInstanceOf[UnknownTopicOrPartitionException])
+      case e: Throwable =>
+        e.printStackTrace()
+        throw e
+    } finally {
+      jsonFile.delete()
+    }
+  }
+
+  /** Test the case where several partitions are given */
+  @Test
+  def testMultiplePartitionsSameAssignment() {
+    val testPartitionA = new TopicPartition("testA", 0)
+    val testPartitionB = new TopicPartition("testB", 0)
+    val testPartitionAssignment = List(1, 2, 0)
+    val testPartitionPreferredLeader = testPartitionAssignment.head
+    val testPartitionAndAssignment = Map(testPartitionA -> testPartitionAssignment, testPartitionB -> testPartitionAssignment)
+
+    createTestTopicAndCluster(testPartitionAndAssignment)
+    bounceServer(testPartitionPreferredLeader, testPartitionA)
+    // Check the leader for the partition is not the preferred one
+    assertNotEquals(testPartitionPreferredLeader, getLeader(testPartitionA))
+    assertNotEquals(testPartitionPreferredLeader, getLeader(testPartitionB))
+    val jsonFile = toJsonFile(testPartitionAndAssignment.keySet)
+    try {
+      PreferredReplicaLeaderElectionCommand.run(Array(
+        "--bootstrap-server", bootstrapServer(),
+        "--path-to-json-file", jsonFile.getAbsolutePath))
+    } finally {
+      jsonFile.delete()
+    }
+    // Check the leader for the partition IS the preferred one
+    assertEquals(testPartitionPreferredLeader, getLeader(testPartitionA))
+    assertEquals(testPartitionPreferredLeader, getLeader(testPartitionB))
+  }
+
+  /** What happens when the preferred replica is already the leader? */
+  @Test
+  def testNoopElection() {
+    createTestTopicAndCluster(testPartitionAndAssignment)
+    // Don't bounce the server. Doublec heck the leader for the partition is the preferred one
+    assertEquals(testPartitionPreferredLeader, getLeader(testPartition))
+    val jsonFile = toJsonFile(testPartitionAndAssignment.keySet)
+    try {
+      // Now do the election, even though the preferred replica is *already* the leader
+      PreferredReplicaLeaderElectionCommand.run(Array(
+        "--bootstrap-server", bootstrapServer(),
+        "--path-to-json-file", jsonFile.getAbsolutePath))
+      // Check the leader for the partition still is the preferred one
+      assertEquals(testPartitionPreferredLeader, getLeader(testPartition))
+    } finally {
+      jsonFile.delete()
+    }
+  }
+
+  /** What happens if the preferred replica is offline? */
+  @Test
+  def testWithOfflinePreferredReplica() {
+    createTestTopicAndCluster(testPartitionAndAssignment)
+    bounceServer(testPartitionPreferredLeader, testPartition)
```

Comment:
Yes, I think so. We need the leader to be other than the preferred one so that a preferred election is possible. With just `createTestTopicAndCluster()` the leader will be the preferred one already, so no election would take place and we wouldn't really be testing what happens if the preferred replica is offline.